jps-plugin/src/org/intellij/erlang/jps/builder/ErlangPrepareDependenciesBuilder.java

```diff
@@ -0,0 +1,270 @@
+                    @NotNull CompileContext context) throws ProjectBuildException, IOException {
+    ErlangDependencyTree dependencyTree = loadDependencyTree(context);
+    if (dependencyTree == null) {
+      addPrepareDependenciesFailedMessage(context);
+      return;
+    }
+
+    List<String> dirtyErlangFilePaths = new DirtyFileProcessor<String, ErlangPrepareDependenciesTarget>() {
+      @Nullable
+      @Override
+      protected String getDirtyElement(@NotNull ErlangPrepareDependenciesTarget target,
+                                       @NotNull File file,
+                                       @NotNull ErlangSourceRootDescriptor root) throws IOException {
+        return isSourceOrHeader(file) ? file.getAbsolutePath() : null;
+      }
+    }.collectDirtyElements(holder);
+    prepareDependencies(target.getProject(), context, dependencyTree, dirtyErlangFilePaths);
+  }
+
+  @NotNull
+  @Override
+  public String getPresentableName() {
+    return NAME;
+  }
+
+  @Nullable
+  private static ErlangDependencyTree loadDependencyTree(@NotNull CompileContext context) {
+    File dataStorageRoot = context.getProjectDescriptor().dataManager.getDataPaths().getDataStorageRoot();
+    File depsTreeFile = new File(dataStorageRoot, DEPENDENCIES_TREE_FILE_PATH);
+    if (!depsTreeFile.exists()) return null;
+    ErlangDependencyTree erlangDependencyTree;
+    try {
+      Document document = JDOMUtil.loadDocument(depsTreeFile);
+      erlangDependencyTree = XmlSerializer.deserialize(document, ErlangDependencyTree.class);
+    }
+    catch (XmlSerializationException e) {
+      return null;
+    }
+    catch (JDOMException e) {
+      return null;
+    }
+    catch (IOException e) {
+      return null;
+    }
+    return erlangDependencyTree;
+  }
+
+  private static boolean isSourceOrHeader(@NotNull File file) {
+    return file.getName().endsWith(".erl") || file.getName().endsWith(".hrl");
+  }
+
+  private static void prepareDependencies(@NotNull JpsProject project,
+                                          @NotNull CompileContext context,
+                                          @NotNull ErlangDependencyTree dependencyTree,
+                                          @NotNull List<String> dirtyErlangFilePaths) {
+    SortedModuleDependencyGraph graph = new SortedModuleDependencyGraph(dependencyTree);
+    Set<String> allDirtyFiles = getAllDirtyFiles(dirtyErlangFilePaths, GraphGenerator.create(graph));
+    ErlangModuleBuildOrders buildOrders = createBuildOrdersForDirtyErlangModules(project, dependencyTree.myNodes, allDirtyFiles);
+    try {
+      Document serializedBuildOrders = new Document(XmlSerializer.serialize(buildOrders, new SkipDefaultValuesSerializationFilters()));
+      File dataStorageRoot = context.getProjectDescriptor().dataManager.getDataPaths().getDataStorageRoot();
+      File depsConfigFile = new File(dataStorageRoot, ErlangBuilder.DEPENDENCIES_CONFIG_FILE_PATH);
```

Comment:
Once again, naming of things is really important.

You're now storing two files - deps-config.xml, and deps-tree.xml under erlang-builder - which one's what? 

DEPENDENCIES_CONFIG_FILE_PATH is used to store dirty modules which are to be rebuilt - it's not what it was used for before.